internal/connector/connector.go

```diff
@@ -305,6 +304,9 @@ func (cw *wrapper) C(ctx context.Context) (types.ConnectorClient, error) {
 
 // Close shuts down the grpc server and closes the connection.
 func (cw *wrapper) Close() error {
+	cw.mtx.Lock()
```

Comment:
tiny: its weird how we have an C() which returns an instance, and caches it.

but then Close() unconditionally closes it.

it feels like maybe it should be a reference counting by default, and a HardClose()/Shutdown() if we're actually serious.

just a thought.